tiledb/sm/query/readers/aggregators/min_max_aggregator.cc

```diff
@@ -0,0 +1,413 @@
+/**
+ * @file min_max_aggregator.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements class MinMaxAggregator.
+ */
+
+#include "tiledb/sm/query/readers/aggregators/min_max_aggregator.h"
+
+#include "tiledb/sm/array_schema/array_schema.h"
+#include "tiledb/sm/query/query_buffer.h"
+#include "tiledb/sm/query/readers/aggregators/aggregate_buffer.h"
+
+using namespace tiledb::common;
+
+namespace tiledb {
+namespace sm {
+
+class MinMaxAggregatorStatusException : public StatusException {
+ public:
+  explicit MinMaxAggregatorStatusException(const std::string& message)
+      : StatusException("MinMaxAggregator", message) {
+  }
+};
+
+template <typename T>
+MinMaxAggregator<T>::MinMaxAggregator(
+    const bool min,
```

Comment:
We can discuss the specifics tomorrow, but this API will need to be changed. If we are going to make this operator-specific, we should not indicate the mode with a bool; _at very least_ the argument here should be an enum, but preferably the functor would also be a template argument with specializations chosen based on the enum. But see discussion below.